lib/p2p/packets/Packet.ts

```diff
@@ -3,41 +3,61 @@ import CryptoJS from 'crypto-js';
 import MD5 from 'crypto-js/md5';
 
 type PacketHeader = {
+  /** The Base64 encoded MD5 hash of the body of this packet. This is used for error checking and as a key to identify packets.*/
   hash: string;
+  /** The hash of the body of the received packet which this packet is responding to. */
   reqHash?: string;
 };
 
-enum MessageType {
+enum PacketDirection {
+  /** A packet that is pushed to a peer without expecting any response. */
   UNILATERAL,
+  /** A packet that is sent to a peer requesting a particular response. */
   REQUEST,
+  /** A packet that is sent in response to an incoming packet. */
   RESPONSE,
 }
 
-abstract class Packet {
-  public header!: PacketHeader;
-  public abstract type: PacketType;
-  public abstract messageType: MessageType;
-  public abstract body: any;
+function createHeader(body: any, reqHash?: string): PacketHeader {
```

Comment:
We were defining functions like this: `const createHeader = (body: any, reqHash?: string): PacketHeader => {` and not with the `function` keyword everywhere else.

This has nothing to do with this specific PR but we are mixing the way we define functions quite a bit.